File: facade.py
class Motor:
    def encender(self):
        print("El motor está encendido.")

    def apagar(self):
        print("El motor está apagado.")

class Luces:
    def encender(self):
        print("Las luces están encendidas.")

    def apagar(self):
        print("Las luces están apagadas.")

class Radio:
    def encender(self):
        print("La radio está encendida.")

    def apagar(self):
        print("La radio está apagada.")

class Automovil:
    def __init__(self):
        self.motor = Motor()
        self.luces = Luces()
        self.radio = Radio()

    def modo_viaje(self):
        print("Activando modo viaje...")
        self.motor.encender()
        self.luces.encender()
        self.radio.encender()

    def modo_estacionado(self):
        print("Activando modo estacionado...")
        self.motor.apagar()
        self.luces.apagar()
        self.radio.apagar()

if __name__ == "__main__":
    auto = Automovil()
    auto.modo_viaje()
    print("---")
    auto.modo_estacionado()

File: java/facade.java
// Clase que representa el Motor
class Motor {
    public void encender() {
        System.out.println("El motor está encendido.");
    }
    
    public void apagar() {
        System.out.println("El motor está apagado.");
    }
}

// Clase que representa las Luces
class Luces {
    public void encender() {
        System.out.println("Las luces están encendidas.");
    }
    
    public void apagar() {
        System.out.println("Las luces están apagadas.");
    }
}

// Clase que representa la Radio
class Radio {
    public void encender() {
        System.out.println("La radio está encendida.");
    }
    
    public void apagar() {
        System.out.println("La radio está apagada.");
    }
}

// Clase que representa el Automóvil y actúa como Facade
class Automovil {
    private Motor motor;
    private Luces luces;
    private Radio radio;
    
    public Automovil() {
        motor = new Motor();
        luces = new Luces();
        radio = new Radio();
    }
    
    public void modoViaje() {
        System.out.println("Activando modo viaje...");
        motor.encender();
        luces.encender();
        radio.encender();
    }
    
    public void modoEstacionado() {
        System.out.println("Activando modo estacionado...");
        motor.apagar();
        luces.apagar();
        radio.apagar();
    }
}

// Clase pública con el método main
public class facade {
    public static void main(String[] args) {
        Automovil auto = new Automovil();
        auto.modoViaje();
        System.out.println("---");
        auto.modoEstacionado();
    }
}
